charts/testkube-cloud-api/templates/migrations-job.yaml

```diff
@@ -51,23 +52,30 @@ spec:
               value: {{ .Values.payments.apiKey }}
               {{- end }}
             {{- end }}
+            {{- $mongoDsnSecretRef := .Values.global.mongo.dsnSecretRef | default .Values.api.mongo.dsnSecretRef }}
+            {{- $mongoDsn := .Values.global.mongo.dsn | default .Values.api.mongo.dsn }}
+            {{- if $mongoDsnSecretRef }}
             - name: API_MONGO_DSN
-              {{- if .Values.api.mongo.dsnSecretRef }}
               valueFrom:
                 secretKeyRef:
                   key: MONGO_DSN
-                  name: {{ .Values.api.mongo.dsnSecretRef }}
-              {{- else }}
-              value: {{ .Values.api.mongo.dsn }}
-              {{- end }}
+                  name: {{ $mongoDsnSecretRef }}
+            {{- else if $mongoDsn }}
+            - name: API_MONGO_DSN
+              value: {{ $mongoDsn }}
+            {{- end }}
+            {{- $mongoDatabase := .Values.global.mongo.database | default .Values.api.mongo.database }}
             - name: API_MONGO_DB
-              value: {{ .Values.api.mongo.database }}
+              value: {{ $mongoDatabase }}
+            {{- $mongoReadPreference := .Values.global.mongo.readPreference | default .Values.api.mongo.readPreference }}
+            - name: API_MONGO_READ_PREFERENCE
+              value: {{ $mongoReadPreference }}
             - name: MIGRATION_IGNORE_NO_DB_ERROR
               value: "{{ .Values.api.migrations.ignoreNoDbError }}"
-            - name: MIGRATION_INIT_INDEXES
-              value: "false"
             - name: MIGRATION_MIGRATE_DATABASE
               value: "true"
+            - name: MIGRATION_INIT_INDEXES
+              value: "true"
```

Comment:
No. It should no longer be needed. I kept the flag though since all the logic was already there. Maybe we have to split it up again for some reason in the future. In hindsight I should probably have cleaned it up.